### PR TITLE
Fix hex case

### PR DESCRIPTION
**Pull request by chriseth**

### CODE REVIEW

File: libdevcore/CommonData.cpp [hunk 1]
```diff
@@ -29,26 +29,49 @@
...
+		ret[i++] = 'x';
+	}
 
 	int rix = _data.size() - 1;
 	for (uint8_t c: _data)
 	{
 		// switch hex case every four hexchars
-		auto hexcase = std::nouppercase;
+		char const* chars = lowerHexChars;
 		if (_case == HexCase::Upper)
```
Comment (ekpyron) [inline libdevcore/CommonData.cpp]: Shouldn't the ``Upper`` and ``Lower`` case be handled outside the loop and only the ``Mixed`` case inside the loop?

**Review by ekpyron — COMMENTED** [review]
I wonder whether we should reconsider just pulling in fmtlib for this kind of stuff...

File: libdevcore/CommonData.cpp (hunk 1 above)
Comment (ekpyron) [reply libdevcore/CommonData.cpp]: Not that it matters too much...

Comment (chriseth) [reply libdevcore/CommonData.cpp]: Sounds good!

**Review by ekpyron — APPROVED** [review]

File: libdevcore/CommonData.cpp [hunk 2]
```diff
@@ -29,26 +29,52 @@
...
+
+string dev::toHex(uint8_t _data, HexCase _case)
+{
+	assertThrow(_case != HexCase::Mixed, BadHexCase, "Mixed case can only be used for byte arrays.");
+
+	char const* chars = _case == HexCase::Upper ? upperHexChars : lowerHexChars;
+
+	return std::string{
+		chars[(unsigned(_data) / 16) & 0xf],
+		chars[unsigned(_data) & 0xf]
```
Comment (axic) [inline libdevcore/CommonData.cpp:48]: Why cast to unsigned when the input is `uint8_t`? Also why not shift by 4 instead of divide by 16?

Comment (ekpyron) [reply libdevcore/CommonData.cpp:48]: I'd expect any of that to result in the same code after compiling anyways...

Comment (axic) [reply libdevcore/CommonData.cpp:48]: Yes, but readability is different :wink

Comment (chriseth) [reply libdevcore/CommonData.cpp:48]: I somehow remembered that `uint8_t`, aka `char` might be weird towards some operators on some platforms. But I could also be wrong about that...

**Pull request merged**